Scripts/Runtime/Entities/TaskDriver/System/AbstractTaskDriverSystem.cs

```diff
@@ -0,0 +1,210 @@
+using Anvil.CSharp.Logging;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Unity.Entities;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    internal abstract partial class AbstractTaskDriverSystem : AbstractAnvilSystemBase,
+                                                               ITaskSetOwner
+    {
+        private static readonly NoOpJobConfig NO_OP_JOB_CONFIG = new NoOpJobConfig();
+        private static readonly List<AbstractTaskDriver> EMPTY_SUB_TASK_DRIVERS = new List<AbstractTaskDriver>();
+
+        private readonly List<AbstractTaskDriver> m_TaskDrivers;
+
+        private BulkJobScheduler<AbstractJobConfig> m_BulkJobScheduler;
+        private bool m_IsHardened;
+        private bool m_IsUpdatePhaseHardened;
+        private bool m_HasCancellableData;
+
+        public AbstractTaskDriverSystem TaskDriverSystem { get => this; }
+
+        public new World World { get; }
```

Comment:
I've added a comment explaining this in more detail but the reason we are hiding it is because we require a valid `World` in the constructor of the system. But because it's a System, we don't get a valid World until `OnCreate`. 

Because `AbstractTaskDriver` controls the creation of a `AbstractTaskDriverSystem` we can inject the `World` in through the constructor and assign it to our new property. This way `AbstractTaskDriverSystem`s are guaranteed to have a valid World right from construction.